providers/fips/fipsprov.c

```diff
@@ -716,6 +719,7 @@ static void fips_teardown(void *provctx)
 {
     OSSL_LIB_CTX_free(PROV_LIBCTX_OF(provctx));
     ossl_prov_ctx_free(provctx);
+    CRYPTO_THREAD_cleanup_local(&rcu_local_key);
```

Comment:
This doesn't work. You can't do this. You may have the fips provider loaded twice into two different lib ctxs. As soon as the first one gets unloaded this will cleanup the shared key and break the second one.